pkg/kube/kube_handler.go

```diff
@@ -801,12 +870,15 @@ func (a *userHandler) GetNamespaces() ([]corev1.Namespace, error) {
 		} else {
 			return nil, err
 		}
-	}
 
-	// Filter namespaces in which the user has permissions to write (secrets) only
-	namespaceList, err := filterAllowedNamespaces(a.clientset, namespaces.Items)
-	if err != nil {
-		return nil, err
+		// Filter namespaces in which the user has permissions to write (secrets) only
+		namespaceList, err = filterAllowedNamespaces(a.clientset, namespaces.Items)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// If the user can list namespaces, do not filter them
+		namespaceList = namespaces.Items
```

Comment:
Ah, so this explains why I thought the FAQ change didn't match the current behaviour - you're changing that behavior. Why? I mean, the fact that a user can list namespaces doesn't mean that it's useful to list all namespaces as if they can use them? I thought that the existing behaviour was like that for that reason.

If the ability to list a namespace was RBAC on an individual namespace, this would make sense, but given it's a cluster perm that just says you can list (all) namespaces regardless of whether you have RBAC to use them, I don't see how the change makes sense? Perhaps you mean that in practise, if people have the cluster RBAC to list namespaces, then they're most likely to have the ability to use them?

OK with me, just want the change to be intentional and understand the reason.

Reply:
No, I believe there was a reason for us doing the filtering even if the user has permissions to list namespaces but I cannot recall why. 

IMO, we don't need to be smarter than the Kubernetes API here. If a user has permissions to list namespaces, why should we hide them? That's how all kubernetes dashboard work: allowing to list namespaces if the user has permissions and then return an error if they try do see/modify something they don't have permissions.

Reply:
Yeah, I'm pretty sure it was because people can have the RBAC to list namespaces without being able to use those namespaces. But yep - I don't mind either way, and it makes sense to list them if they have perms to do so, as you say.